doc/ui_component2_transition.md

```diff
@@ -0,0 +1,135 @@
+# UiComponent2 Transition Notes
+> A document to explain the new features and breakages when converting components from Component / UiComponent to 
+Component2 / UiComponent2.
+---
+* __[What's New in Component2](#whats-new-in-component2)__
+* __[Why a new base class?](#why-a-new-base-class)__
+* __[New Features, improvements, and breakages](#new-features-improvements-and-breakages)__
+    * [Small Perf Improvements](#small-perf-improvements)
+    * [JS-Backed Maps](#js-backed-maps)
+    * [Lifecycle Method Changes](#lifecycle-method-changes)
+    * [Other Method Changes](#other-method-changes)
+    * [PropTypes](#prop-validation-goes-through-proptypes)
+    * [ErrorBoundary Component Added](#errorboundary-component-added)
+    * [Other Breaking Changes](#other-breakages-we-made-while-we-had-the-opportunity)
+* __[Updating](#updating)__
+## What's New in Component2
+`Component` and `UiComponent` are being deprecated in favor of new versions of the API: `Component2` and `UiComponent2`.
```

Comment:
Since this document lives in the repo, I think it'd be helpful to add a little bit of context as to when this occurred.

Something like:
```suggestion
In `react` 5.1.0 and `over_react` 3.1.0, `Component` and `UiComponent` were deprecated in favor of new versions of the API: `Component2` and `UiComponent2`.
```